src/api/bookstore.js

```diff
@@ -0,0 +1,50 @@
+const BASE_URL = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi';
+
+export const initialize = async () => {
+  if (!localStorage.getItem('appId')) {
+    const response = await fetch(`${BASE_URL}/apps/`, {
+      method: 'POST',
+    });
+    const appId = await response.text();
+    localStorage.setItem('appId', appId);
+  }
+};
+
+export const getBooks = async () => {
+  const appId = localStorage.getItem('appId');
+  const response = await fetch(`${BASE_URL}/apps/${appId}/books`);
+  const books = await response.text();
+  return books;
+};
+
+export const addNewBook = async (book) => {
+  const appId = localStorage.getItem('appId');
+  const response = await fetch(`${BASE_URL}/apps/${appId}/books`, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({
+      item_id: book.id,
+      title: `${book.title}&|&|&${book.author}`,
+      category: book.category,
+    }),
+  });
+  const status = await response.text();
+  return status;
```

Comment:
- [ ] The `localstorage` is set but it's not storing anything. see screenshot 👇

![bnbjkbjlv](https://user-images.githubusercontent.com/61562626/130690105-95bbd60b-74fa-478e-8aa0-9eb01af9d1ee.png)

Reply:
Well done here.

![react4](https://user-images.githubusercontent.com/55635977/130749241-d5deb8fc-db0a-4302-a1b8-7e2d9fa88978.png)